### PR TITLE
Add CHAOS reward to current FeeManager

### PR DESCRIPTION
**Pull request by hippocampus-web3**
## Release Notes

- **New Features**
	- Introduced a `chaos` token to enhance fee management and reward claiming.
	- Added functionality for owners to withdraw `chaos` tokens and set conversion rates.
	- New event logging for rewards claimed and rewards converted to USDC, improving transparency for users.

- **Bug Fixes**
	- Enhanced validation for constructor parameters in the `FeeManager` to prevent invalid addresses.

- **Tests**
	- Expanded testing coverage for the `FeeManager` and `ArrakisV2` contracts, including scenarios for minting, burning, and fee distribution, as well as edge cases related to rate settings and withdrawals.

### CODE REVIEW

File: contracts/FeeManager.sol [hunk 1]
```diff
@@ -100,10 +121,19 @@ contract FeeManager is IFeeManager, Ownable {
             return;
         }
 
-        // TODO: Add event
         rewardDebt[claimer] = totalReward;
 
+        uint256 rewardsToHarvestInChaos = (rewardsToHarvest *
+            rate *
+            10**(_CHAOS_DECIMALS - _USDC_DECIMALS)) / 100;
```
Comment (bitcoinbrisbane) [inline contracts/FeeManager.sol:134]: Maybe just check _CHAOS_DECIMALS > _USDC_DECIMALS.  I guess it always should be.

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM.  Just double check the code rabbit suggestions
